forge/csrc/passes/erase_inverse_ops.cpp

```diff
@@ -336,12 +339,25 @@ void commute_and_bypass(graphlib::Graph *graph, std::vector<graphlib::Node *> co
                     }
                 }
                 op->set_shape(op_shape);
-                update_reshape_attr(op, op_shape);
+                if (op->op_name() == "reshape")
+                    update_reshape_attr(op, op_shape);
             }
 
             auto [in_edge, out_edge] = insert_node_on_edge(graph, operand_edge, clone);
             // Set dataformat to match producer on operand edge
             clone->set_output_df(graph->node_by_id(in_edge.producer_node_id)->output_df());
+
+            // Remove braodcast tms from clone's data edges and recalculate shapes
+            std::vector<graphlib::Edge> clone_data_edges = graph->operand_data_edges(clone);
```

Comment:
Why are we removing broadcast TMs here?

Reply:
Hi @nvukobratTT ,
Broadcast tms are removed only for edges which has consumer as squeeze or unsqueeze ops. If we have broadcast tms for theses edges, op shape is miscalculated, which is the root cause for this issue at MLIR stage.

As adding broadcast tms for squeeze & unsqueeze op is not valid, this is removed now.